File: src/main/java/io/github/jakejmattson/mousemeasure/MouseMeasure.java
package io.github.jakejmattson.mousemeasure;

import org.jnativehook.GlobalScreen;
import org.jnativehook.NativeHookException;
import org.jnativehook.mouse.NativeMouseEvent;
import org.jnativehook.mouse.NativeMouseInputListener;

import java.awt.*;
import java.util.logging.Level;
import java.util.logging.Logger;

public class MouseMeasure implements NativeMouseInputListener {

    private Point origin;
    private TextDisplay topLeft = new TextDisplay();
    private TextDisplay bottomRight = new TextDisplay();

    public static void main(String[] args)
    {
        disableLogging();
        MouseMeasure example = new MouseMeasure();

        try {
            GlobalScreen.registerNativeHook();
            GlobalScreen.addNativeMouseListener(example);
            GlobalScreen.addNativeMouseMotionListener(example);
        }
        catch (NativeHookException ex) {
            System.err.println("There was a problem registering the native hook.");
            System.err.println(ex.getMessage());
            System.exit(1);
        }
    }

    private static void disableLogging()
    {
        Logger logger = Logger.getLogger(GlobalScreen.class.getPackage().getName());
        logger.setLevel(Level.WARNING);
        logger.setUseParentHandlers(false);
    }

    public void nativeMouseDragged(NativeMouseEvent e)
    {
        Point current = e.getPoint();
        bottomRight.setLocation(current);
        bottomRight.setText("X: " + e.getX() + " Y: " + e.getY());
        bottomRight.setVisible(true);
    }

    public void nativeMousePressed(NativeMouseEvent e)
    {
        origin = e.getPoint();
        topLeft.setLocation(origin);
        topLeft.setText("X: " + e.getX() + " Y: " + e.getY());
        topLeft.setVisible(true);
    }

    public void nativeMouseReleased(NativeMouseEvent e)
    {
        topLeft.setVisible(false);
        bottomRight.setVisible(false);
    }

    public void nativeMouseClicked(NativeMouseEvent e) {}
    public void nativeMouseMoved(NativeMouseEvent e) {}
}

File: src/main/java/io/github/jakejmattson/mousemeasure/MouseMeasure.kt
package io.github.jakejmattson.mousemeasure

import org.jnativehook.*
import org.jnativehook.mouse.*
import java.awt.Point
import java.util.logging.*

class MouseMeasure : NativeMouseInputListener {
    private val topLeft = TextDisplay(Location.TOP_LEFT)
    private val bottomRight = TextDisplay(Location.BOTTOM_RIGHT)

    override fun nativeMouseDragged(e: NativeMouseEvent) = updateFrame(bottomRight, e.point)
    override fun nativeMousePressed(e: NativeMouseEvent) = updateFrame(topLeft, e.point)

    override fun nativeMouseReleased(e: NativeMouseEvent) {
        topLeft.setVisible(false)
        bottomRight.setVisible(false)
    }

    override fun nativeMouseClicked(e: NativeMouseEvent) {}
    override fun nativeMouseMoved(e: NativeMouseEvent) {}

    private fun updateFrame(display: TextDisplay, location: Point) {
        display.setLocation(location)
        display.setText(pointToString(location))
        display.setVisible(true)
    }

    private fun pointToString(point: Point): String {
        return "X: " + point.x + " Y: " + point.y
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            disableLogging()
            val example = MouseMeasure()
            try {
                GlobalScreen.registerNativeHook()
                GlobalScreen.addNativeMouseListener(example)
                GlobalScreen.addNativeMouseMotionListener(example)
            } catch (e: NativeHookException) {
                e.printStackTrace()
            }
        }

        private fun disableLogging() {
            val logger = Logger.getLogger(GlobalScreen::class.java.getPackage().name)
            logger.level = Level.WARNING
            logger.useParentHandlers = false
        }
    }
}